### ISSUE TITLE
Custom arguments under argument groups are not shown in the web UI

### ISSUE THREAD

**Issue by cyberw** [locustio/locust#2560]
### Prerequisites

- [X] I am using [the latest version of Locust](https://github.com/locustio/locust/releases/)
- [X] I am reporting a bug, not asking a question

### Description

I would love for custom arguments to become visible in the web ui, even when they are under an argument group. Right now, for example, none of locust-plugins's arguments are visible.

Ideally they would be shown under their separate group titles.

Doesnt need to be fixed in the "old" UI (as long as it doesnt break it). Pretty soon I feel we are ready to switch the default to modern anyway.

### Command line

.

### Locustfile contents

```python3
@events.init_command_line_parser.add_listener
def add_arguments(parser: LocustArgumentParser):
    checks = parser.add_argument_group(
        "locust-plugins - Checks",
        "Sets locust's exit code to 3 if any of these thresholds were not met",
    )
    checks.add_argument(
        "--check-rps",
        type=float,
        help="Requests per second",
        env_var="LOCUST_CHECK_RPS",
    )
```

### Python version

.

### Locust version

latest

### Operating system

.

**Comment by cyberw** [comment]
@andrewbaldwin44 Could you have a look, when you have the time?

**Comment by andrewbaldwin44** [comment]
Yeah it seems to be more of an issue with the argument parser, rather than the web UI :) The `extra_options` that are displayed in the Web UI are compiled in the `argument_parser.ui_extra_args_dict` method. The arguments are then filtered such that only arguments with the property `include_in_web_ui=True` are shown.

Although the property `include_in_web_ui` is defaulted to be `True`, it is only defaulted to be `True` when an argument is added through the custom `add_argument` method, which is not the case for argument groups.

Overriding the `add_argument_group` proves to be a bit tricky or weird. I'll try to come up with a solution or maybe a different way to filter the args. Then to have them grouped by argument group we will also need to keep track of the group they belong to, which as far as I can tell is not done by default

**Comment by cyberw** [comment]
Aha, I think I can solve the `add_argument_group` part.

**Comment by cyberw** [comment]
Hmm. My approach didnt work. Happy to see someone else have a go at it..

**Comment by pranavthombare** [comment]
I'm also looking into it. Maybe I share the working code snippet here once I'm able to get it to work.
